numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -593,6 +593,30 @@ fail:
     return NULL;
 }
 
+/*
+ * Helper to test fromstring of 0 terminated strings, as the C-API supports
+ * the -1 length identifier.
+ */
+static PyObject *
+fromstring_null_term_c_api(PyObject *dummy, PyObject *byte_obj)
+{
+    char *string;
+
+    if (!PyBytes_CheckExact(byte_obj)) {
+        PyErr_SetString(PyExc_ValueError, "argument must be a byte string.");
```

Comment:
Shouldn't this be a TypeError?

Reply:
In fact, since this is just test code - can you skip these lines entirely and let  `PyBytes_AsString` throw the error?